### PR TITLE
Fixes for building with Ryzen AI Software 1.3.0

### PR DESCRIPTION
**Pull request by fifield**
Two changes here:

1) The Ryzen AI Software 1.3.0 installer installs things into a python venv. The binaries are put in the venv binary path (e.g. `venv/bin`) whereas the libraries and headers are installed into the python site-packages directory (e.g.`venv/lib/python3.10/site-packages/include`). This means that just using the path of `xchesscc` is not adequate for finding `AIETOOLS_DIR` path. So, this PR updates FindAIETools.cmake to search for `adf.h` to locate the include path instead of inferring it from the binary path. Then, `AIETOOLS_DIR` path is the parent of the include dir.

2) Make DSP lib optional because it is not present in the Ryzen AI software installation.

### CODE REVIEW

**Comment by fifield** [comment]
Without this [I get errors](https://github.com/fifield/mlir-aie/actions/runs/11243558802/job/31259719824#step:4:2024) like:
```
  [139/591] Generating mm.o
  FAILED: aie_kernels/mm.o /workspace/build/aie_kernels/mm.o 
  cd /workspace/build/aie_kernels && /workspace/mlir-aie/tools/chess-clang/xchesscc_wrapper AIE2 -aietools /workspace/venv -I /workspace/venv/include -c /workspace/mlir-aie/aie_kernels/mm.cc -o /workspace/build/aie_kernels/mm.o
  Error: Couldn't open "/workspace/venv/data/aie_ml/lib/me.prx": no such file or directory
  xchesscc Failed 
  The Program xchesscc has encountered an unexpected error exiting ... xchesscc Failed 
  ```

File: FindAIETools.cmake
[the remark this answers is not in the document]
Comment (denolf) [reply FindAIETools.cmake:1]: and same for AIETOOLS_INCLUDE_DIR please

[the remark this answers is not in the document]
Comment (denolf) [reply FindAIETools.cmake:1]: and can you update the header as well with the new variables (lines 39-44)
Thanks!

**Review by denolf — CHANGES_REQUESTED** [review]

File: FindAIETools.cmake
[the remark this answers is not in the document]
Comment (fifield) [reply FindAIETools.cmake:1]: Resolved in https://github.com/Xilinx/cmakeModules/pull/30/commits/2bf8bf92b26fd467a0bbfff8b76bbde72d5edc55

[the remark this answers is not in the document]
Comment (denolf) [reply FindAIETools.cmake:1]: thanks

**Review by denolf — APPROVED** [review]

**Pull request merged**